From dfdc1874b09e06d47369d33c2b6bad9f0d664c5c Mon Sep 17 00:00:00 2001
From: pdknsk <pdknsk@gmail.com>
Date: Mon, 03 Mar 2014 21:46:19 +0000
Subject: Assume sRGB when image doesn't have an ICC profile attached

This seems to be no problem with other image viewers and
is apparently also recommended by W3C.

https://bugzilla.gnome.org/show_bug.cgi?id=554498
---
diff --git a/src/eog-image.c b/src/eog-image.c
index 2b646f5..d94604a 100644
--- a/src/eog-image.c
+++ b/src/eog-image.c
@@ -693,7 +693,14 @@ eog_image_apply_display_profile (EogImage *img, cmsHPROFILE screen)
 
 	priv = img->priv;
 
-	if (screen == NULL || priv->profile == NULL) return;
+	if (screen == NULL) return;
+
+	if (priv->profile == NULL) {
+		/* Assume sRGB color space for images without ICC profile */
+		eog_debug_message (DEBUG_LCMS, "Image has no ICC profile. "
+		                   "Assuming sRGB.");
+		priv->profile = cmsCreate_sRGBProfile ();
+	}
 
 	/* TODO: support other colorspaces than RGB */
 	if (cmsGetColorSpace (priv->profile) != cmsSigRgbData ||
--
cgit v0.9.2
